pkg/cmd/build/build.go

```diff
@@ -0,0 +1,443 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package build
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/images/archive"
+	"github.com/containerd/containerd/platforms"
+	dockerreference "github.com/containerd/containerd/reference/docker"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/buildkitutil"
+	"github.com/containerd/nerdctl/pkg/clientutil"
+	"github.com/containerd/nerdctl/pkg/platformutil"
+	"github.com/containerd/nerdctl/pkg/strutil"
+	"github.com/sirupsen/logrus"
+)
+
+func Build(ctx context.Context, options *types.BuildCommandContext) error {
+	platform := options.Platform
+	buildctlBinary, buildctlArgs, needsLoading, metaFile, tags, cleanup, err := generateBuildctlArgs(ctx, options)
+	if err != nil {
+		return err
+	}
+	if cleanup != nil {
+		defer cleanup()
+	}
+
+	quiet := options.Quiet
+
+	logrus.Debugf("running %s %v", buildctlBinary, buildctlArgs)
+	buildctlCmd := exec.Command(buildctlBinary, buildctlArgs...)
+	buildctlCmd.Env = os.Environ()
+
+	var buildctlStdout io.Reader
+	if needsLoading {
+		buildctlStdout, err = buildctlCmd.StdoutPipe()
+		if err != nil {
+			return err
+		}
+	} else {
+		buildctlCmd.Stdout = options.StdOut
+	}
+	if !quiet {
+		buildctlCmd.Stderr = options.StdErr
+	}
+
+	if err := buildctlCmd.Start(); err != nil {
+		return err
+	}
+
+	if needsLoading {
+		platMC, err := platformutil.NewMatchComparer(false, platform)
+		if err != nil {
+			return err
+		}
+		if err = loadImage(ctx, buildctlStdout, options.GOptions.Namespace, options.GOptions.Address, options.GOptions.Snapshotter, options.StdOut, platMC, quiet); err != nil {
+			return err
+		}
+	}
+
+	if err = buildctlCmd.Wait(); err != nil {
+		return err
+	}
+
+	iidFile := options.IidFile
+	if iidFile != "" {
+		id, err := getDigestFromMetaFile(metaFile)
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(iidFile, []byte(id), 0600); err != nil {
+			return err
+		}
+	}
+	namespace := options.GOptions.Namespace
+
+	address := options.GOptions.Address
```

Comment:
If they are used only once after assignment, we can remove these code and use `Options.GOptions.Namespace/Address` directly.